packages/dataviews/src/types.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+type Item = Record< string, any >;
+
+interface Option {
+	value: any;
+	label: string;
+}
+
+interface filterByConfig {
+	operators?: string[];
+	isPrimary?: boolean;
+}
+
+export interface Field {
+	/**
+	 * The unique identifier of the field.
+	 */
+	id: string;
+
+	/**
+	 * The label of the field. Defaults to the id.
+	 */
+	header?: string;
+
+	/**
+	 * Callback used to retrieve the value of the field from the item.
+	 * Defaults to `item[ field.id ]`.
+	 */
+	getValue?: ( { item }: { item: Item } ) => any;
+
+	/**
+	 * Callback used to render the field. Defaults to `field.getValue`.
+	 */
+	render?: ( { item }: { item: Item } ) => ReactNode;
+
+	/**
+	 * The width of the field column.
+	 */
+	width?: string | number;
```

Comment:
I'd like us to remove the width properties from the Field API. The `view.layout` may be a good fit.